aiida/orm/nodes/repository.py

```diff
@@ -0,0 +1,297 @@
+            warnings.warn('keyword `key` changed to `path`.', AiidaDeprecationWarning)  # pylint: disable=no-member
+            path = key
+
+        return self._repository.list_object_names(path)
+
+    @contextlib.contextmanager
+    def open(self, path: str, mode='r', key: str = None) -> io.BufferedReader:
+        """Open a file handle to an object stored under the given key.
+
+        .. note:: this should only be used to open a handle to read an existing file. To write a new file use the method
+            ``put_object_from_filelike`` instead.
+
+        .. deprecated:: 1.4.0
+            The keyword `key` will be removed in `v2.0.0`, it has been renamed to `path`.
+
+        :param path: the relative path of the object within the repository.
+        :return: yield a byte stream object.
+        :raises TypeError: if the path is not a string and relative path.
+        :raises FileNotFoundError: if the file does not exist.
+        :raises IsADirectoryError: if the object is a directory and not a file.
+        :raises OSError: if the file could not be opened.
+        """
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn('keyword `key` changed to `path`.', AiidaDeprecationWarning)  # pylint: disable=no-member
+            path = key
+
+        if mode not in ['r', 'rb']:
+            raise ValueError('the mode {} is not supported.'.format(mode))
+
+        with self._repository.open(path) as handle:
+            if 'b' not in mode:
+                yield io.StringIO(handle.read().decode('utf-8'))
+            else:
+                yield handle
+
+    def get_object_content(self, path: str, mode='r', key: str = None, encoding='utf-8') -> typing.Union[str, bytes]:
+        """Return the content of a object identified by key.
+
+        .. deprecated:: 1.4.0
+            The keyword `key` will be removed in `v2.0.0`, it has been renamed to `path`.
+
+        :param key: fully qualified identifier for the object within the repository.
+        :raises TypeError: if the path is not a string and relative path.
+        :raises FileNotFoundError: if the file does not exist.
+        :raises IsADirectoryError: if the object is a directory and not a file.
+        :raises OSError: if the file could not be opened.
+        """
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn('keyword `key` changed to `path`.', AiidaDeprecationWarning)  # pylint: disable=no-member
+            path = key
+
+        if mode not in ['r', 'rb']:
+            raise ValueError('the mode {} is not supported.'.format(mode))
+
+        if 'b' not in mode:
+            return self._repository.get_object_content(path).decode(encoding)
+
+        return self._repository.get_object_content(path)
+
+    def put_object_from_filelike(
```

Comment:
No longer accepts the `force` keyword (probably the same for other methods).

Reply:
Thanks, I have added a deprecation for these.